README.md

```diff
@@ -243,6 +243,49 @@ These *internal keys* are
 - cls: the concrete class of a pickled object
 - val: identifies the fields of a pickled object
 
+##Solving chicken and egg pickling problem
+
+Situation of having  circular dependencies is very common and makes pickling a little bit complicating.
+Consider the following trait Hierarchy.
+
+```scala
+
+    trait Rule
+    trait ArcRule(propertyName:String) extends Rule
+
+    trait AndRule(label:String,rules:Rule*) extends Rule
+```
+
+If you define picklers in a common way then you will get into trouble
+
+```scala
+  implicit val rulePickler: PicklerPair[Rule] = CompositePickler[Rule]
+    .concreteType[ArcRule].concreteType[AndRule]
+
+  implicit val andPickler: Pickler[AndRule] = Pickler.materializePickler[AndRule]
```

Comment:
@antonkulaga Ive got some time to look over this topic proposal. Would like to understand problem since I haven't yet hit it myself. 

But as is, this code example is not possible, since `Pickler.materializePickler[AndRule]` won't work: materialize only works on case classes and objects. Can you clarify please?